### PR TITLE
Add timer to check connection between OCU and JRB

### PR DESCRIPTION
**Pull request by patapufate**
# Description
If the Jaus Ros Bridge loses connection with the OCU, The mux will still sending commands to the Mux. Because the Jaus Ros Bridge has more priority than the Autopilot, the vehicle won't receive commands from the mission control / Autopilot.

Taking into accout that the OCU sends "Manual Command" to the Jaus Ros Brids every 1 second, this PR adds a timer in the Jaus Ros Bridge that check the connection with OCU. 
## Type of change

Please mark options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings

# How To Test
To check if the timer is working, the connection between the Jaus Ros Bridge and the OCU must be cut manually.

### CODE REVIEW

File: jaus_ros_bridge/include/JausDataManager.h [hunk 1]
```diff
@@ -95,6 +95,8 @@ class JausDataManager {
   void handleReportRPM(const thruster_control::ReportRPM::ConstPtr& msg);
 
  private:
+  void checOCUConnection(const ros::TimerEvent& ev);
```
Comment (hidmic) [inline jaus_ros_bridge/include/JausDataManager.h:98]: @patapufate typo: 
```suggestion
  void checkOCUConnection(const ros::TimerEvent& ev);
```

File: jaus_ros_bridge/include/JausDataManager.h [hunk 2]
```diff
@@ -114,7 +116,9 @@ class JausDataManager {
   udpserver* _udp;
   ros::Publisher _publisher_EnableLogging;
   ros::Publisher _publisher_ClearFault;
+  ros::Timer _checkOCUCommunication;
```
Comment (hidmic) [inline jaus_ros_bridge/include/JausDataManager.h:119]: @patapufate nit: 
```suggestion
  ros::Timer _timer_checkOCUCommunication;
```

**Review by hidmic — APPROVED** [review]
I'm OK with this for a quick fix, but I do wonder whether we'd be better off with a state-less bridge. Which means the OCU has to publish thruster and fins messages at a faster rate. CC @ChrisScianna for feedback.

File: jaus_ros_bridge/include/JausDataManager.h (hunk 1 above)
Comment (patapufate) [reply jaus_ros_bridge/include/JausDataManager.h]: fixed!

File: jaus_ros_bridge/include/JausDataManager.h (hunk 2 above)
Comment (patapufate) [reply jaus_ros_bridge/include/JausDataManager.h]: Fixed

**Comment by ChrisScianna** [comment]
So two things:

It takes about 5 seconds to stop after disconnecting. This may be a limitation with the detection of loss of comms in which case we will have to just deal with it.  

If it reconnects it starts going the speed it disconnected at.

**Comment by ChrisScianna** [comment]
This fixed the reconnect error where it started again after reconnecting.

Current Bug: The vehicle now drops to 0 rpm and immediately back to commanded every 3 seconds when connected.

File: jaus_ros_bridge/include/PowerPlantControl.h [hunk 3]
```diff
@@ -78,8 +78,8 @@ class PowerPlantControl : public JausMessageIn {
   ros::Publisher _publisher_setRPM;
   ros::Timer _commandRPMTimer;
   int _maxRPM;
-
   bool _isSetToZero;
+
```
Comment (hidmic) [inline jaus_ros_bridge/include/PowerPlantControl.h:82]: @patapufate nit: unnecessary blank line changes.

File: jaus_ros_bridge/src/PowerPlantControl.cpp [hunk 4]
```diff
@@ -52,7 +52,6 @@ void PowerPlantControl::init(ros::NodeHandle* nodeHandle) {
       _nodeHandle->advertise<thruster_control::SetRPM>("input/jaus_ros_bridge/set_rpm", 1, true);
 
   _isSetToZero = true;
-
```
Comment (hidmic) [inline jaus_ros_bridge/src/PowerPlantControl.cpp:55]: @patapufate nit: unrelated change

File: jaus_ros_bridge/src/PowerPlantControl.cpp [hunk 5]
```diff
@@ -119,16 +118,18 @@ int PowerPlantControl::GetRpm() {
...
 
 void PowerPlantControl::PublishRPM(const bool enable)
 {
   if (enable)
     _commandRPMTimer.start();
-  else
+  else{
     _commandRPMTimer.stop();
+    StopThruster();
+  }
```
Comment (hidmic) [inline jaus_ros_bridge/src/PowerPlantControl.cpp:131]: @patapufate nit: use braces for both clauses:

```c++
  if (enable) {
    _commandRPMTimer.start();
  } else {
    _commandRPMTimer.stop();
    StopThruster();
  }
```

Comment (hidmic) [reply jaus_ros_bridge/src/PowerPlantControl.cpp:131]: I'm not thrilled by this "double" stop, but OK for now.

**Review by hidmic — APPROVED** [review]

File: jaus_ros_bridge/include/PowerPlantControl.h (hunk 3 above)
Comment (patapufate) [reply jaus_ros_bridge/include/PowerPlantControl.h:82]: Fixed

File: jaus_ros_bridge/src/PowerPlantControl.cpp (hunk 4 above)
Comment (patapufate) [reply jaus_ros_bridge/src/PowerPlantControl.cpp]: fixed
